### ISSUE TITLE
Add RCV data entry interface for audit boards

### ISSUE THREAD

**Issue by mcchilders** [votingworks/arlo#1308]
**Description:**
For audits of the RAIRE RCV Ballot Comparison type, the audit boards should see a slightly different ballot data entry interface that allows them to indicate ranked votes for various candidates. See screenshot here:
![Screen Shot 2021-10-19 at 11.46.48 AM.png](https://zube.io/files/votingworks/2df89fb57b370e7c1ca790b2a29cda14-screen-shot-2021-10-19-at-11-46-48-am.png)

Changes to this interface from our normal one include:
- The addition of the ranking dropdown box after each candidate's name, with ranking options equal to x in the "vote for = x" of the contest settings
- Clicking the candidate name/choice buttons should not do anything by itself in RCV audits - the candidate/choice must be selected by setting the rank, at which point the button will color in/activate. 
-- Note: if it's easy to make these not look like/function like buttons for this feature, that's fine too - they can just be a list of names
- The blank/null state for the ranking options should have lighter-colored placeholder text that says "No rank"
- The ranked options should be in the format "1st choice/2nd choice/3rd choice" but, if possible, it would be great to also be able to click in the box and type 1/2/3 to choose those wordier options
- Auditors MAY record an overvote (e.g. put "2nd choice" in two dropdowns OR use shift/select to select multiple ranks in a single dropdown) - that is not an error
- Auditors MAY leave some of the rankings blank and not others - that is not an error
- Audits MAY NOT leave all of the rankings blank unless they also click the "blank vote" button on the right

**Comment by mcchilders** [comment]
@cjerdonek yep, it depends on the jurisdiction's counting rules - but good catch, because we do need to allow multi-select in the dropdowns since, for your example, Rank 1 would count but Rank 2 would be an overvote -> exhausted ballot under SFs rules.

@MorganLove note the addition ^^ in the description; a shift/select should allow you to select multiple ranks within a single dropdown (which is an edge case/error, but one we need to account for.)

**Comment by cjerdonek** [comment]
@mcchilders Another idea occurred to me that I think would make data entry a lot easier and may also simplify the implementation, including removing the need for multi-select and displaying and storing multi-select, etc. You could add an "Overvote" choice below "Write-in." Then, in my example, it would be Sandy 1st and Overvote 2nd. Data enterers would only need to indicate the highest ranking for each candidate and, if there is an overvote, the highest ranking at which an overvote occurs. One reason this would simplify data entry is that e.g. if someone ranks Sandy 1st, 2nd, 3rd, 4th, 5th (e.g. at all rankings), which isn't uncommon, data enterers wouldn't feel like they need to select all 5 since the UI wouldn't permit it.

**Comment by mcchilders** [comment]
@cjerdonek you're right that it might simplify a lot of things, but one of the core principles in RLAs is that you want auditors to enter what they see on the ballot, the votes the voters actually marked (subject to whatever voter intent guidelines are used by the jurisdiction, of course); you explicitly *don't* want auditors interpreting those voter marks according to how they think the tabulator would have/should have counted them. So we never ask anyone to decide, for example, that something is an overvote, and we wouldn't want them to only mark some votes and not others (e.g. only the first ranking.) 

Re: terminology for multiple rankings for the same candidate, I'm sure you're probably right, but I haven't found another term for it anywhere (and I've looked!)   Let me know if you come across one. 

Given that the instructions given to SF RCV voters are: 
- only select one candidate in each column, and
- always select a different candidate in each column

I feel ok calling both of those overvote conditions (although the first exhausts the ballot and the other just skips you to the next ranking under SFs rules).

**Comment by cjerdonek** [comment]
@mcchilders Okay, thanks. That's a valid position regarding the manual entry. Out of curiosity, how would the audit be affected if the ballot says Sandy 1st and 2nd, but the data enterer only enters Sandy 1st? Could it result in more ballots needing to be sampled because it's a strict mismatch, or would it never cause more ballots to be sampled because it would never affect any of the vote totals?

**Comment by umbernhard** [comment]
Also, it's worth noting that the UI mocked up above is quite out of date with our current AB UI: 

![Image](https://user-images.githubusercontent.com/2686765/228022439-25de8efa-04d9-49d5-b43d-4cab822a7711.png)
